client/my-sites/stats/stats-purchase/test/test-use-available-upgrade-tiers.js

```diff
@@ -0,0 +1,65 @@
+import {
+	extendTiersBeyondHighestTier,
+	transformTiers,
+	filterLowerTiers,
+} from '../use-available-upgrade-tiers';
+import originalTiers from './tiers-fixture';
+
+const tiers = transformTiers( originalTiers );
+const CURRENCY_CODE = 'USD';
+
+describe( 'extendTiersBeyondHighestTier', () => {
+	it( 'should return original tiers if not purchased or with little monthly views', () => {
+		const usageData = { views_limit: null, billableMonthlyViews: 0 };
+		const extendedTiers = extendTiersBeyondHighestTier( tiers, CURRENCY_CODE, usageData );
+
+		expect( extendedTiers.length ).toBe( 6 );
+		expect( extendedTiers[ 0 ] ).toEqual( tiers[ 0 ] );
+		expect( extendedTiers[ 1 ] ).toEqual( tiers[ 1 ] );
+	} );
+	it( 'should return original tiers if purchased', () => {
+		const usageData = { views_limit: 10000, billableMonthlyViews: 0 };
+		const newTiers = filterLowerTiers( tiers, usageData );
```

Comment:
we should test those functions separately. Or just test the top level function. Will do tomorrow.